src/main/java/com/google/sps/data/NSWGovAdaptor.java

```diff
@@ -0,0 +1,163 @@
+    return siteCoord;
+  }
+
+
+  private void updateSiteInfo() throws Exception {
+    //Creating a get request
+    URL url = new URL("https://data.airquality.nsw.gov.au/api/Data/get_SiteDetails");
+    HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+    connection.setRequestMethod("GET");
+    
+    connection.setRequestProperty("Content-Type", "application/json");
+    
+    int responseCode = connection.getResponseCode();
+
+    ArrayList<GovSiteDetails> convertedlist = new ArrayList<GovSiteDetails>();
+    
+    System.out.println(java.time.LocalDate.now());  
+
+    try {
+      if (responseCode == HttpURLConnection.HTTP_OK) {
+        BufferedReader br = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+        StringBuilder response = new StringBuilder();
+        String responseLine; 
+        while ((responseLine = br.readLine()) != null) {
+          response.append(responseLine.trim());
+        }
+  
+        String responseString = response.toString();
+        Gson gson = new Gson();
+        convertedlist = gson.fromJson(responseString, new TypeToken<ArrayList<GovSiteDetails>>() {}.getType());
+
+        //Convert the arrayList into a map. 
+        for(GovSiteDetails details : convertedlist){
+          dataMap.put(details.siteId , details);
+        } 
+      } else {
+
+        //Throw custom exception when the response code is not 200
+        throw new HTTPStatusCodeException("HTTP Status Code is not 200");
+      }
+    } catch (final Exception e) {
+      System.out.println("updateSiteInfo getMessage(): " + e.getMessage()); 
+    }
+  }
+
+  //Post request to extract the AQI from the government API, given a desired date
+  private ArrayList<NSWGovAQDataPoint> extractAQI(LocalDate inputDate) throws Exception {
+    final URL url = new URL("https://data.airquality.nsw.gov.au/api/Data/get_Observations");
+    final HttpURLConnection con = (HttpURLConnection) url.openConnection();
+    con.setRequestMethod("POST");
+
+    //Set "content-type" request header to "application/json" to send the request content in JSON form
+    //Charset encoding as uT8 which is default, useful if the request enconding is diff to UT8 encoding
+    con.setRequestProperty("Content-Type", "application/json; utf-8");
+
+    //Set request header to application/json to read the response 
+    con.setRequestProperty("Accept", "application/json");
+
+    //Enable DoOutput to send request content and write content to output stream
+    con.setDoOutput(true);
+
+    String desiredDate = inputDate.format(DateTimeFormatter.ofPattern("yyyy-MM-dd"));
+    System.out.println(LocalDate.now());
+
+    //Create the Request Body 
+    final String jsonInputString = "{" +
+      "\"Parameters\": [ \"AQI\", ]," +
+      "\"Sites\": [ ]," +
+      String.format("\"StartDate\": \"%s\",", desiredDate) +
+      String.format("\"EndDate\": \"%s\",", desiredDate) + 
```

Comment:
same above, will it work with startDate & endDate set to the same day?

Reply:
yes